tools/cache.py

```diff
@@ -9,6 +9,8 @@ def __init__(self, dirname=None, debug=False):
       dirname = os.environ.get('EM_CACHE')
     if not dirname:
       dirname = os.path.expanduser(os.path.join('~', '.emscripten_cache'))
+      if os.environ.get('EMCC_WASM_BACKEND') and os.environ.get('EMCC_WASM_BACKEND') != '0':
+        dirname += '_wasm'
```

Comment:
Could we use a subdirectory inside the cache? Sibling directories are difficult, since users may presume there is just a single cache directory they need to interact with. Also, looks like this only does it if `EM_CACHE` is not being used, which is active for emsdk installations. Could we always use a structure `cache_root/asmjs/...` and `cache_root/wasm/...` if we want totally separate ones?